build.gradle

```diff
@@ -47,6 +47,9 @@ repositories {
 }
 
 dependencies {
+    // We should not include javax.mail in the dependencies as it is included with App Engine SDK and adding it will cause
+    // conflicts. However, in a very unlikely scenario that it is required, we should repackage it or use other workarounds.
+    // See https://github.com/TEAMMATES/teammates/pull/7525 for more info.
```

Comment:
I think I do not really like the idea of leaving the comment in the codebase either. However, if we don't leave this here, people may come in to add the dependency without realizing the conflicts.

The other option is that we should have a dependency management docs, explaining what each dependency is doing, why we are including it, and how they are linked together with other dependencies. I think that is very important but we do not have any at the moment. And instead of having the document in a separate place, I think the best place to place it is within `build.gradle` itself.

So I understand this comment looks out of place but as we start to incrementally add dependency management documentation, it would look better fitting.

Of course, we could always have a separate document on dependency management, but in-code documentation seems to be preferable to me. If we do have a separate document on dependency management though, that means I could include docs on possible workarounds for conflicting dependencies, but I don't want anyone modifying build.gradle to miss that document.

How about a short comment in the dependencies block, and then we start to slowly grow a dependency management document?

For example,

`dependencies {
     // When adding/ modifying the dependencies, please make sure you have read docs/depedency-management.md.
...
}